server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+                    //printf("malloc_succ\n");
+                }
+                syslog(LOG_INFO, "Accepts connection from %s", inet_ntoa(((struct sockaddr_in*)&their_addr)->sin_addr));
+        }
+        int recv_len = 0;
+        
+        while (accepted)
+        {   
+            /* code */
+            int i = 0;
+            recv_len = recv(sock_accept_fd,(void*)rec_val,REC_LEN,0);
+            if(recv_len == -1)
+            {
+                perror("recv_error");
+                syslog(LOG_ERR, "Recv ERROR");
+                //printf("recv_error\n");
+                exit(EXIT_FAILURE);
+            }   
+            else if(recv_len == 0)//Connection failed
+            {
+                //printf("In recvlen 0\n");
+                accepted = 0;
+                total_length = 0;
+                break;
+            }           
+            while(i < recv_len)
+            {
+                if(rec_val[i] == '\n')
+                {
+                    push_data = 1;
+                    i++;
+                    break;
+                }
+                i++;    
+            }
+            store_data = (char*)realloc(store_data,total_length+i);
+            memcpy(store_data+total_length,rec_val,i);
+            total_length+=i;
+                
+            memset(rec_val,0,recv_len);
+            if(recv_len !=0 && push_data == 1)
+            {
+                
+                //My data is stored in the buffer, now write it to the file file_aesdsocket
+                file_fd = open(file_aesdsocket,O_WRONLY|O_APPEND);
+                if(file_fd == -1)
+                {
+                    printf("File opening failed\n");
+                    syslog(LOG_ERR, "File opening failed");
+                    exit(EXIT_FAILURE);
+                }
+                //printf("length is %d\n",total_length);
+                int nr = write(file_fd, store_data, total_length);
+                if(nr == -1)
+	            {
+		            //Log error if print failed
+		            syslog(LOG_ERR, "write is not sucessful");
+                    //printf("Write_failed\n");
+		            close(file_fd);
+		            exit(EXIT_FAILURE);
+	            }
+	            else
+	            {
+	                //check if the whole string is written to the  file
+		            if(nr == total_length)
+		            {
+		                syslog(LOG_DEBUG, "Write_Success");
+		                //close(file_fd);
+		            }
+		            else
+		            {
+		        	    syslog(LOG_ERR, "Incorrect information written.Repeat the process");
+                        //printf("Incorrect info written\n");
+		        	    close(file_fd);
+		        	    exit(EXIT_FAILURE);
+		            }
+	            }
+                close(file_fd);
+                //Incrementing file length so that the whole data is pushed
+                file_length+=total_length;
+                char file_data[file_length];
```

Comment:
@SrirajVemparala what does this assume about the size of the stack relative to the total size of all packets sent?